src/test/java/com/googleinterns/zoomtube/servlets/IconFeedbackServletTest.java

```diff
@@ -112,4 +129,61 @@ public void doPost_validRequest_shouldStoreInDatabase() throws IOException, Serv
     assertThat(iconFeedback.timestampMs()).isEqualTo(456);
     assertThat(iconFeedback.type().toString()).isEqualTo("GOOD");
   }
+
+  @Test
+  public void doGet_missingLectureId_shouldRespondWithBadRequest()
+      throws IOException, ServletException {
+    servlet.doGet(request, response);
+
+    verify(response).sendError(
+        HttpServletResponse.SC_BAD_REQUEST, /* message= */ "Missing lecture id parameter.");
+  }
+
+  @Test
+  public void doGet_noFeedback_shouldReturnNoIconFeedback() throws IOException, ServletException {
+    when(request.getParameter(IconFeedbackServlet.PARAM_LECTURE_ID)).thenReturn("123");
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json");
+    List<IconFeedback> result = getIconFeedbackFromJson(content.toString());
+    assertThat(result.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void doGet_oneIconFeedback_shouldReturnOneIconFeedback()
+      throws IOException, ServletException {
+    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, 123);
+    datastoreService.put(IconFeedbackUtil.createEntity(lectureKey, 456L, IconFeedback.Type.GOOD));
+    when(request.getParameter(IconFeedbackServlet.PARAM_LECTURE_ID)).thenReturn("123");
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json");
+    List<IconFeedback> result = getIconFeedbackFromJson(content.toString());
+    assertThat(result.size()).isEqualTo(1);
+    assertThat(result.get(0).lectureKey().getId()).isEqualTo(123);
+    assertThat(result.get(0).timestampMs()).isEqualTo(456);
+    assertThat(result.get(0).type()).isEqualTo(IconFeedback.Type.GOOD);
+  }
+
+  @Test
+  public void doGet_oneIconFeedbackLectureIdDoesntMatch_shouldReturnNoIconFeedback()
+      throws IOException, ServletException {
+    Key lectureKey = KeyFactory.createKey(LectureUtil.KIND, 123);
+    datastoreService.put(IconFeedbackUtil.createEntity(lectureKey, 456L, IconFeedback.Type.GOOD));
+    when(request.getParameter(IconFeedbackServlet.PARAM_LECTURE_ID)).thenReturn("789");
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json");
+    List<IconFeedback> result = getIconFeedbackFromJson(content.toString());
+    assertThat(result.size()).isEqualTo(0);
+  }
+
+  private List<IconFeedback> getIconFeedbackFromJson(String json) {
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    Type listType = new TypeToken<ArrayList<IconFeedback>>() {}.getType();
```

Comment:
I thought there was another/better way we did this?

Reply:
Talked with Nathan offline, weird bug

Reply:
NVM

Reply:
I think that is how I currently have it as well.